mirgecom/euler.py

```diff
@@ -20,16 +22,214 @@
 THE SOFTWARE.
 """
 
-# import numpy as np
-# import numpy.linalg as la  # noqa
-# from pytools.obj_array import (
-#         join_fields, make_obj_array,
-#         with_object_array_or_scalar)
-# from meshmode.mesh import BTAG_ALL, BTAG_NONE  # noqa
-# from grudge.eager import with_queue
-# from grudge.symbolic.primitives import TracePair
+import numpy as np
+import numpy.linalg as la  # noqa
+from pytools.obj_array import (
+    flat_obj_array,
+    make_obj_array,
+    with_object_array_or_scalar,
+)
+import pyopencl.clmath as clmath
+import pyopencl.array as clarray
+from meshmode.mesh import BTAG_ALL, BTAG_NONE  # noqa
+from mirgecom.boundary import DummyBoundary
+from mirgecom.eos import IdealSingleGas
+
+from grudge.eager import with_queue
+from grudge.symbolic.primitives import TracePair
+
+# from grudge.dt_finding import (
+#    dt_geometric_factor,
+#    dt_non_geometric_factor,
+# )
+
+__doc__ = """
+.. autofunction:: inviscid_operator
+"""
+
+
+#
+# Euler flow eqns:
+# d_t(q) + nabla .dot. f = 0 (no sources atm)
+# state vector q: [rho rhoE rhoV]
+# flux tensor f: [rhoV (rhoE + p)V (rhoV.x.V + p*I)]
+#
+
+
+def _interior_trace_pair(discr, vec):
+    i = discr.interp("vol", "int_faces", vec)
+    e = with_object_array_or_scalar(
+        lambda el: discr.opposite_face_connection()(el.queue, el), i
+    )
+    return TracePair("int_faces", i, e)
+
+
+def _inviscid_flux(discr, q, eos=IdealSingleGas()):
+
+    ndim = discr.dim
+
+    # q = [ rho rhoE rhoV ]
+    mass = q[0]
+    energy = q[1]
+    mom = q[2:]
+
+    p = eos.pressure(q)
+
+    # Fluxes:
+    # [ rhoV (rhoE + p)V (rhoV.x.V + p*I) ]
+    momflux = make_obj_array(
+        [
+            (mom[i] * mom[j] / mass + (p if i == j else 0))
+            for i in range(ndim)
+            for j in range(ndim)
+        ]
+    )
+    massflux = mom * make_obj_array([1.0])
+    energyflux = mom * make_obj_array([(energy + p) / mass])
+
+    flux = flat_obj_array(massflux, energyflux, momflux,)
+
+    return flux
```

Comment:
You do this "assign to a variable, then return" a lot. Could you clean those up?

Reply:
Typically these are leftover from debugging where I was trying to inspect stuff.  I'll gettem out.